pkg/deploy/api/validation/validation_test.go

```diff
@@ -365,6 +365,40 @@ func TestValidateDeploymentConfigMissingFields(t *testing.T) {
 			fielderrors.ValidationErrorTypeRequired,
 			"template.strategy.rollingParams.pre.failurePolicy",
 		},
+		"small template.strategy.rollingParams.updatePercent": {
```

Comment:
Refactored the tests slightly to support valid cases. Upper/lower bound validity and errors are now tested.